tests/ssh_tests.py

```diff
@@ -83,13 +85,23 @@ def trying_to_use_ssh_shell_after_exit_results_in_error():
     assert_raises(Exception, lambda: shell.run(["true"]))
 
 
-def _create_shell_with_wrong_port():
+@istest
+def passing_a_custom_socket_like_a_pre_connected_one():
+    sock = socket.socket(socket.AddressFamily.AF_INET, socket.SOCK_STREAM)
+    retry_on_signal(lambda: sock.connect((HOSTNAME, PORT)))
+
+    with _create_shell_with_wrong_port(sock=sock) as shell:
+        shell.run(["true"])
+
+
+def _create_shell_with_wrong_port(sock=None):
     return spur.SshShell(
-        username="bob",
-        password="password1",
-        hostname="localhost",
+        username=USERNAME,
+        password=PASSWORD,
+        hostname=HOSTNAME,
         port=54321,
         missing_host_key=spur.ssh.MissingHostKey.accept,
+        sock=sock
```

Comment:
well if i want to connect to 2.2.2.2 via 1.1.1.1
i will do 

SshShell(hostname="2.2.2.2", sock=ProxyCommand("ssh 1.1.1.1 nc 2.2.2.2 22")

So the proxy command string contains both the hop and the target because the nc (netcat) command needs it